doc/source/data/examples/batch_inference_object_detection.ipynb

```diff
@@ -0,0 +1,350 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2023-05-09 16:47:15,553\tINFO streaming_executor.py:91 -- Executing DAG InputDataBuffer[Input] -> TaskPoolMapOperator[ReadImage]\n",
+      "2023-05-09 16:47:15,554\tINFO streaming_executor.py:92 -- Execution config: ExecutionOptions(resource_limits=ExecutionResources(cpu=None, gpu=None, object_store_memory=None), locality_with_output=False, preserve_order=False, actor_locality_enabled=True, verbose_progress=False)\n",
+      "2023-05-09 16:47:15,554\tINFO streaming_executor.py:94 -- Tip: For detailed progress reporting, run `ray.data.DataContext.get_current().execution_options.verbose_progress = True`\n"
+     ]
+    },
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "08e680739a224c7d8a173e3adaa9061c",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "Running 0:   0%|          | 0/200 [00:00<?, ?it/s]"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "2023-05-09 16:47:18,180\tINFO streaming_executor.py:149 -- Shutting down <StreamingExecutor(Thread-1089, started daemon 13592719360)>.\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "(4,) (478, 612, 3)\n"
+     ]
+    }
+   ],
+   "source": [
+    "single_batch = ds.take_batch(batch_size=4)\n",
+    "print(single_batch[\"image\"].shape, single_batch[\"image\"][0].shape)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Batch inference with Ray Data\n",
+    "\n",
+    "Then we will demostrate how to use Ray Data to do batch inference in a distributed cluster.\n",
+    "\n",
+    "The first step is to package the model in a Python class. The class is mainly consist of two parts. In the `__init__` constructor, we package the code that loads and initializes the model. And in the `__call__` method, we package the code that applies model on each batch."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from typing import Dict\n",
+    "\n",
+    "import numpy as np\n",
+    "from torchvision import transforms\n",
+    "from torchvision.models.detection import (FasterRCNN_ResNet50_FPN_V2_Weights,\n",
+    "                                          fasterrcnn_resnet50_fpn_v2)\n",
+    "\n",
+    "\n",
+    "class ObjectDetectionModel:\n",
+    "    def __init__(self):\n",
+    "        # Define the model loading and initialization code in `__init__`\n",
+    "        self.weights = FasterRCNN_ResNet50_FPN_V2_Weights.DEFAULT\n",
```

Comment:
why are we using FasterRCNN model?

Reply:
This is from PyTorch's official example: https://pytorch.org/vision/stable/models.html#object-detection

Reply:
Yep, can we mention that in the example!

Reply:
yeah, already linked.